File: c/p2.c
#include <stdio.h>

int fibonacci(int a, int b, int n)
{
  if(n == 0)
  {
    return a;
  }

  return fibonacci(b, a + b, n - 1);
}

int main(int argc, const char * argv[])
{
  int i = 0;
  int fib = 0;
  long sum = 0L;

  while(fib < 4000000)
  {
    fib = fibonacci(0, 1, i);
    if(fib % 2 == 0)
    {
      sum += (long) fib;
    }
    i++;
  }

  printf("%ld\n", sum);

  return 0;
}

File: julia/p2.jl
function fib(a, b, n)
  if n == 0
    return a
  end
  return fib(b, a + b, n - 1)
end

i = 0
f = 0
s = 0
while(f < 4000000)
  f = fib(0, 1, i)
  if f % 2 == 0
    s += f
  end
  i += 1
end
println(s)